### PR TITLE
[GraphQL] Fix pruning compatibility issues

### PR DESCRIPTION
**Pull request by amnn**
## Description 

Two further fixes to get GraphQL working with pruned databases.

## Test plan 

Tested by deploying this version of the service against a pruned DB, and ensuring the following query executed successfully:

```graphql
query {
  chainIdentifier
  protocolConfig(protocolVersion: 10) {
    featureFlags {
      key
      value
    }
  }
}
```

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [x] GraphQL: Adds support for running GraphQL against a pruned version of the service.
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: crates/sui-indexer/src/schema.rs [hunk 1]
```diff
@@ -286,14 +286,11 @@ macro_rules! for_all_tables {
             checkpoints,
             epochs,
             events,
-            events_partition_0,
```
Comment (wlmyng) [inline crates/sui-indexer/src/schema.rs:289]: was this automatically or manually removed?

**Review by wlmyng — APPROVED** [review]
:ok:

**Review by stefan-mysten — APPROVED** [review]

**Review by emmazzz — APPROVED** [review]
triple approve!

File: crates/sui-indexer/src/schema.rs (hunk 1 above)
Comment (emmazzz) [reply crates/sui-indexer/src/schema.rs:289]: Probably manually since partition 0's were pruned away?

Comment (amnn) [reply crates/sui-indexer/src/schema.rs:289]: Manually (but this part is already updated manually).

**Pull request merged**